### ISSUE TITLE
[rush] Rush doesn't like Node.js version 10.13+ even though it is LTS

### ISSUE THREAD

**Issue by ScottHysom** [Microsoft/web-build-tools#1134]
Since Node.js version 10.13.0+ has been LTS since 2018-10-30, can Rush be updated to no longer complain about this being unsupported? What is the process for that?

**Comment by octogonz** [comment]
@ScottHysom 
Are you sure you're using the latest Rush?  I was able to run "rush install" in **web-build-tools** using NodeJS 10.15.0 and I didn't see any errors.

Check your `rushVersion` in [rush.json](https://rushjs.io/pages/configs/rush_json/).  Also check that your globally installed Rush is current.

**Comment by ScottHysom** [comment]
@octogonz When I type `rush --version` or `rush -v`, it says this:
```
> rush --version
Your version of Node.js (10.14.1) has not been tested with this release of Rush. The Rush team will not accept issue reports for it. Please consider upgrading Rush or downgrading Node.js.


Rush Multi-Project Build Tool 5.5.4 - https://rushjs.io


usage: rush [-h] [-d] <command> ...
rush: error: too few arguments
```

That leads me to think I have version 5.5.4. But it turns out that is only reflecting the version in my repo's `rush.json` file. When I look at the globally installed version of rush, its `package.json` has a `"_resolved"` value of 5.0.5 (`https://registry.npmjs.org/@microsoft/rush/-/rush-5.0.5.tgz`). 

So I guess I am running an older version of Rush. Shouldn't the version returned be 5.0.5 instead of what is in rush.json?

I just corroborated this by running rush from outside the repo. It returns 5.0.5.

**Comment by octogonz** [comment]
I agree the user experience could be more clear. Also we need to document the "Rush version selector" feature better.

This probably got overlooked because the people who maintain repos tend to be Rush experts already and know how to keep Rush updated based on rote memorization heheh. Should be easy to fix.

**Comment by ScottHysom** [comment]
Actually, this issue is not about an out-of-date `rushVersion` in rush.json. That is already the most recent 5.5.4. The problem is the instance of Rush that is run is the globally installed version which is older, yet it doesn't complain that it is older than the stated `rushVersion` in rush.json, nor does it say its true version number (it always reflects the `rushVersion`). 

It seems the more accurate thing to reflect is 2 version numbers:
1. The version of Rush that is currently running (which should never be influenced by rush.json).
2. The version of Rush that rush.json requires.

Thoughts?

**Comment by octogonz** [comment]
The concept of the "version selector" is that we want builds to be deterministic, so every person who builds a repo should use the exact same release of Rush. But it would be annoying if your globally installed `rush` tool had to be an exact version, since you may be swtiching between several different repo folders/branches with different requirements.

So the design is that the `rush` command is a simple stub that examines `rushVersion` from rush.json, downloads and installs that version, and then invokes it. This means e.g. that the `rush -h` command-line help that you see may be different in different folders, since it may be using different Rush releases.

As such, the globally installed package version really doesn't matter too much, since the relevant bootstrapping code is small and rarely changes. It would normally only matter if some issue was fixed regarding how we fetch and install Rush.

But the NodeJs version check is an obvious exception, and I suppose has to hapoen as early as possible during startup.

But yeah, we should warn when things are out of date. Last time we discussed this, people were concerned about Rush "phoning home" to the NPM public registry during a build, so we considered a passive check like this:

- If the globally installed Rush is newer than rush.json by a minor version, then warn that rush.json should be upgraded (thinking being that some developer will eventually come along with a newer Rush and encounter the warning)
- If the global Rush version is older than rush.json, then warn the person that they should upgrade their tool
- When showing the warning about outdated NodeJs, we could also suggest to check whether your Rush is outdated

Would that address your issue? It's pretty easy to code.

**Comment by ScottHysom** [comment]
That explanation is helpful, thanks @octogonz. Your proposal listed in the bullet points makes sense and I think would address the issue.
